bin/smart-proxy-openscap-send

```diff
@@ -37,10 +37,25 @@ if !Proxy::SETTINGS.foreman_url
   exit 437
 end
 
+lockfile = File.join(loaded_settings.spooldir, ".spool.lock")
+
+if File.exist? lockfile
+  logger.debug "Lock file #{lockfile} for openscap spool exists, not sending spool to server"
+  exit
+end
+
 begin
+  FileUtils.touch lockfile
   Proxy::OpenSCAP::send_spool_to_foreman(loaded_settings)
 rescue StandardError => e
   logger.error e
   puts "#{e} See #{Proxy::OpenSCAP.fullpath(loaded_settings.openscap_send_log_file)}"
   exit 438
+ensure
+  FileUtils.rm lockfile
 end
+
+Signal.trap("TERM") {
+  FileUtils.rm lockfile
+  exit
+}
```

Comment:
This block must be moved just under line 40, this way it is never executed so the trap will not be active during execution.

Additionally, lockfile can be not yet created, so please check if it exists and delete it.